### ISSUE TITLE
Better heuristics for when nodes should give up training

### ISSUE THREAD

**Issue by bendecoste** [OpenMined/Grid#49]
In the MNIST demo, a common problem happens when running with 3 nodes, which is that one of the nodes starts to be unable to positively contribute to training.

Worker A and worker B start lowering the loss together, and they send their model to worker C, who's data makes the loss go up, so he doesn't publish.

If worker C ran more epochs or changed his training rates, he'd probably be able to lower the error rate.

Basically, the current scheme to decide if a model has improved is very naive.  We should discuss how to make it better!
